### PR TITLE
Pub/Sub: add publish retry sample

### PR DESCRIPTION
**Pull request by anguillanneuf**
This is a new sample that will go on [here](https://cloud.google.com/pubsub/docs/publisher#retry), next to the Java and Node samples.

### CODE REVIEW

**Comment by anguillanneuf** [comment]
@hongalex Please ignore Cloud Build test failure.

File: pubsub/cloud-client/publisher.py [hunk 1]
```diff
@@ -171,17 +172,17 @@ def callback(f):
             try:
                 print(f.result())
                 futures.pop(data)
-            except: # noqa
+            except:  # noqa
```
Comment (hongalex) [inline pubsub/cloud-client/publisher.py]: nit: is this space required by the linter?

Comment (anguillanneuf) [reply pubsub/cloud-client/publisher.py]: Yah, double space is required for single line comments.

File: pubsub/cloud-client/publisher.py [hunk 2]
```diff
@@ -233,36 +292,47 @@ def publish_messages_with_batch_settings(project_id, topic_name):
...
+    )
     publish_with_custom_attributes_parser.add_argument('topic_name')
 
     publish_with_futures_parser = subparsers.add_parser(
-        'publish-with-futures',
-        help=publish_messages_with_futures.__doc__)
-    publish_with_futures_parser.add_argument('topic_name')
+        'publish-with-futures', help=publish_messages_with_futures.__doc__
+    )
+    publish_with_futures_parser.add_argument("topic_name")
```
Comment (hongalex) [inline pubsub/cloud-client/publisher.py]: nit: single quotes

File: pubsub/cloud-client/publisher.py [hunk 3]
```diff
@@ -115,12 +115,13 @@ def publish_messages_with_custom_attributes(project_id, topic_name):
...
     for n in range(1, 10):
-        data = u'Message number {}'.format(n)
+        data = u"Message number {}".format(n)
         # Data must be a bytestring
-        data = data.encode('utf-8')
+        data = data.encode("utf-8")
         # Add two attributes, origin and username, to the message
         future = publisher.publish(
-            topic_path, data, origin='python-sample', username='gcp')
+            topic_path, data, origin="python-sample", username="gcp"
```
Comment (hongalex) [inline pubsub/cloud-client/publisher.py]: Nit: single quotes

File: pubsub/cloud-client/publisher.py [hunk 4]
```diff
@@ -115,12 +115,13 @@ def publish_messages_with_custom_attributes(project_id, topic_name):
     topic_path = publisher.topic_path(project_id, topic_name)
 
     for n in range(1, 10):
-        data = u'Message number {}'.format(n)
+        data = u"Message number {}".format(n)
         # Data must be a bytestring
-        data = data.encode('utf-8')
+        data = data.encode("utf-8")
```
Comment (hongalex) [inline pubsub/cloud-client/publisher.py]: Nit: single quotes

File: pubsub/cloud-client/publisher.py [hunk 5]
```diff
@@ -115,12 +115,13 @@ def publish_messages_with_custom_attributes(project_id, topic_name):
     topic_path = publisher.topic_path(project_id, topic_name)
 
     for n in range(1, 10):
-        data = u'Message number {}'.format(n)
+        data = u"Message number {}".format(n)
```
Comment (hongalex) [inline pubsub/cloud-client/publisher.py]: Nit: single quotes

File: pubsub/cloud-client/publisher.py [hunk 6]
```diff
@@ -223,7 +224,65 @@ def publish_messages_with_batch_settings(project_id, topic_name):
...
+def publish_messages_with_retry_settings(project_id, topic_name):
+    """Publishes messages with custom retry settings."""
+    # [START pubsub_publisher_retry_settings]
+    from google.cloud import pubsub_v1
+
+    # TODO project_id = "Your Google Cloud Project ID"
+    # TODO topic_name = "Your Pub/Sub topic name"
+
+    # Configure the retry settings. Defaults will be overwritten.
+    retry_settings = {
```
Comment (hongalex) [inline pubsub/cloud-client/publisher.py]: I think these should also be using single quotes.

**Pull request merged**